device/mellanox/x86_64-mlnx_msn2700-r0/plugins/psuutil.py

```diff
@@ -64,11 +104,76 @@ def get_psu_presence(self, index):
         if index is None:
             return False
 
-        status = 0
-        try:
-            with open(self.psu_path + self.psu_presence.format(index), 'r') as presence_status:
-                status = int(presence_status.read())
-        except IOError:
-            return False
+        status = self._read_file(self.psu_presence, index)
 
         return status == 1
+
+    def get_output_voltage(self, index):
+        """
+        Retrieves the ouput volatage in milli volts of a power supply unit (PSU) defined
+                by 1-based index <index>
+        :param index: An integer, 1-based index of the PSU of which to query o/p volatge
+        :return: An integer, value of o/p voltage in mV if PSU is good, else zero
+        """
+        if index is None:
+            return 0
```

Comment:
suggest adding an INVALID_VOLTAGE, similar comments to below get_current, get_power, and get_fan_speed API.

Reply:
for values like voltage, current and power, 0 has the meaning that the power doesn't work. In this sense, to define an extra constant looks like overdesign to me.
how do you think?